General improvements

bfcoder opened the pull request:
* Use a yaml file instead of hard coding usernames
* Support rate limiting with an exponential backoff
* Support removing a user if they don't have access to the repo

bfcoder committed:
Use a yaml file instead of hard coding usernames

bfcoder committed:
Support rate limiting with an exponential backoff

bfcoder committed:
Support removing a user if they don't have access to the repo

seanrcollings commented on lib/pt_to_github/migrator.rb:
```diff
@@ -23,13 +23,25 @@ def run
+          elsif e.class == Octokit::UnprocessableEntity && e.detailed_message.match?("422 - Validation Failed")
+            value = JSON.parse(e.response_body)["errors"][0]["value"]
+            puts "The mapped user (#{value}) doesn't have access to the repository. Removing that person."
+            story.owned_by -= [@user_map.key(value)]
```
I think this will end up removing the users from the description of the story, but you'd just have to memoize the description and it'll remain the same across multiple calls

bfcoder replied:
ahh, good point. I'll do that.

bfcoder replied:
Ok, I fixed the description. Tested and it now keeps all original users in the description. Thanks for pointing this out!

bfcoder committed:
Memoize the description

Doing this so removing users from the owned_by array doesn't affect the original description. We want to keep all original users in the description.

bfcoder committed:
Staying consistent with wording
